Mathlib/Analysis/Asymptotics/Asymptotics.lean

```diff
@@ -1926,12 +1926,18 @@ theorem isLittleO_const_id_cobounded (c : F'') :
   isLittleO_const_left.2 <| .inr tendsto_norm_cobounded_atTop
 #align asymptotics.is_o_const_id_comap_norm_at_top Asymptotics.isLittleO_const_id_cobounded
 
-theorem isLittleO_const_id_atTop (c : E'') : (fun _x : ℝ => c) =o[atTop] id :=
-  isLittleO_const_left.2 <| Or.inr tendsto_abs_atTop_atTop
+theorem isLittleO_const_id_cocompact [ProperSpace F''] (c : E'') :
+    (fun _ => c) =o[cocompact F''] id :=
+  isLittleO_const_left.2 <| Or.inr tendsto_norm_cocompact_atTop
+
+theorem isLittleO_const_id_atTop [LinearOrder F''] [NoMaxOrder F''] [ClosedIciTopology F'']
+    [ProperSpace F''] (c : E'') : (fun _ : F'' => c) =o[atTop] id :=
```

Comment:
Do you have some specific examples of `F'' \ne Real` in mind? Could you please mention them in the commit message and/or comment?

Reply:
This seems to work:

```lean
import Mathlib

example : (fun (_: ℤ) ↦ (1 : ℚ)) =o[Filter.atTop] id := Asymptotics.isLittleO_const_id_atTop _
```

(oddly, replacing ℤ with ℚ doesn't work although mathematically it seems like it should)

To be honest though, I didn't have any particular motivation for making this change. I think I wanted the cocompact statement and that happened to pull in the `ProperSpace` assumption which got carried over to the `atBot` and `atTop` versions.